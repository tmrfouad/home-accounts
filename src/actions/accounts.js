import database from '../firebase/firebase';

// Add_ACCOUNT
export const addAccount = account => ({
  type: 'ADD_ACCOUNT',
  account
});

export const startAddAccount = (accountData = {}) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    const { name = '' } = accountData;

    const account = {
      name
    };

    return database
      .ref(`users/${uid}/accounts`)
      .push(account)
      .then(ref => {
        dispatch(addAccount({ id: ref.key, ...account }));
      })
      .catch(() => {});
  };
};

// REMOVE_ACCOUNT
export const removeAccount = ({ id }) => ({
  type: 'REMOVE_ACCOUNT',
  id
});

export const startRemoveAccount = ({ id } = {}) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    return database
      .ref(`users/${uid}/accounts/${id}`)
      .remove()
      .then(() => {
        dispatch(removeAccount({ id }));
      });
  };
};

// EDIT_ACCOUNT
export const editAccount = ({ id } = {}, updates) => ({
  type: 'EDIT_ACCOUNT',
  id,
  updates
});

export const startEditAccount = ({ id } = {}, updates) => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    return database
      .ref(`users/${uid}/accounts/${id}`)
      .update(updates)
      .then(() => {
        dispatch(editAccount({ id }, updates));
      });
  };
};

// SET_ACCOUNTS
export const setAccounts = accounts => ({
  type: 'SET_ACCOUNTS',
  accounts
});

export const startSetAccounts = () => {
  return (dispatch, getState) => {
    const uid = getState().auth.uid;
    return database.ref(`users/${uid}/accounts`).once('value', snap => {
      const accounts = [];
      snap.forEach(childSnap => {
        accounts.push({
          id: childSnap.key,
          ...childSnap.val()
        });
      });
      dispatch(setAccounts(accounts));
    });
  };
};
