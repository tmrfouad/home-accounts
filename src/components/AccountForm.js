import React from 'react';
import { connect } from 'react-redux';
import {
  startAddAccount,
  startEditAccount,
  startRemoveAccount
} from '../actions/accounts';
import ConfirmModal from './ConfirmModal';

export class AccountForm extends React.Component {
  constructor(props) {
    super(props);
    const account = this.props.account;
    this.state = {
      name: account ? account.name : '',
      error: '',
      mode: account ? 'edit' : 'add',
      isModalOpen: false
    };
  }

  onNameChange = e => {
    this.setState({ name: e.target.value });
  };

  onFormSubmit = e => {
    e.preventDefault();
    if (!this.state.name) {
      this.setState({ error: 'Please enter a name!' });
    } else {
      if (this.state.mode === 'edit') {
        this.props.startEditAccount(this.props.account.id, {
          name: this.state.name
        });
      } else {
        if (this.props.accounts.find(acc => acc.name === this.state.name)) {
          this.setState({ error: 'This account already exists!' });
        } else {
          this.props.startAddAccount({
            name: this.state.name
          });
        }
      }
      this.props.history.push('/accounts');
    }
  };

  removeAccount = () => {
    this.props.startRemoveAccount(this.props.account.id);
    this.props.history.push('/accounts');
  };

  openRemoveItemDialog = e => {
    e.preventDefault();
    this.setState(() => ({ isModalOpen: true }));
  };

  closeRemoveItemDialog = () => {
    this.setState(() => ({ isModalOpen: false }));
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h2 className="page-header__title">
              {this.state.mode === 'edit' ? 'Edit' : 'Add'} Account
            </h2>
          </div>
        </div>
        <form className="form content-container" onSubmit={this.onFormSubmit}>
          {this.state.error && (
            <div className="form__error">{this.state.error}</div>
          )}
          <input
            type="text"
            className="text-input"
            value={this.state.name}
            onChange={this.onNameChange}
          />
          <div>
            <button className="button form__action">Save Account</button>
            <button
              className="button form__action button--secondary"
              onClick={this.openRemoveItemDialog}
            >
              Remove Account
            </button>
          </div>
        </form>
        <ConfirmModal
          id="confirmModal"
          messageTitle="Remove Account!"
          messageBody="Are you sure you want to remove this item?"
          isOpen={this.state.isModalOpen}
          onModalClose={this.closeRemoveItemDialog}
          onModalOk={this.removeAccount}
        />
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  account: state.accounts.find(t => t.id === props.match.params.id),
  accounts: state.accounts
});

const mapDispatchToProps = dispatch => ({
  startAddAccount: account => dispatch(startAddAccount(account)),
  startEditAccount: (id, account) =>
    dispatch(startEditAccount({ id }, account)),
  startRemoveAccount: id => dispatch(startRemoveAccount({ id }))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AccountForm);
