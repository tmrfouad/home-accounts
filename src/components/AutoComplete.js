import React from 'react';

export default class AutoComplete extends React.Component {
  constructor(props) {
    super(props);

    if (
      !props.id ||
      !props.source ||
      !props.displayField ||
      !props.valueField ||
      !props.onChange
    ) {
      throw new Error(
        'You must supply the id, src, displayField, valueField and onChange properties for the AutoComplete component.'
      );
    }
  }

  updateInputs() {
    const selectedItem =
      this.props.source.find(
        itm => itm[this.props.valueField] + '' === this.props.value + ''
      ) || {};

    const textInput = document.getElementById(this.props.id + '-text');
    const hiddenInput = document.getElementById(this.props.id + '-hidden');

    textInput.value = selectedItem[this.props.displayField] || '';
    hiddenInput.value = selectedItem[this.props.valueField] || '';
  }

  componentDidUpdate() {
    this.updateInputs();
  }

  componentDidMount() {
    this.updateInputs();
    this.autocomplete();
  }

  autocomplete() {
    const self = this;
    const inp = document.getElementById(self.props.id + '-text');
    const src = self.props.source;
    const displayField = self.props.displayField;
    const valueField = self.props.valueField;
    const onChange = self.props.onChange;

    let currentFocus;

    inp.addEventListener('click', function() {
      showList(true);
    });

    inp.addEventListener('blur', function(e) {
      if (e.target.closest('.autocomplete')) {
        return;
      }
      let inputValue = inp.value;
      const foundObj = src.find(o => o[displayField] + '' === inputValue + '');
      if (foundObj) {
        const value = foundObj[valueField];
        onChange(value);
        closeAllLists();
      } else {
        if (isListOpen()) {
          const firstListItem = inp.parentNode.querySelector(
            '.autocomplete-active'
          );
          if (firstListItem) {
            const value = firstListItem.getElementsByTagName('input')[0].value;
            const firstObj = src.find(
              itm => itm[valueField] + '' === value + ''
            );
            onChange(value);
            closeAllLists();
          }
        }
      }
    });

    /*execute a function when someone writes in the text field:*/
    inp.addEventListener('input', function(e) {
      showList();
    });
    /*execute a function presses a key on the keyboard:*/
    inp.addEventListener('keydown', function(e) {
      let x = document.getElementById(this.id + 'autocomplete-list');
      if (x) x = x.getElementsByTagName('div');
      if (e.keyCode == 40) {
        if (isListOpen()) {
          /*If the arrow DOWN key is pressed,
          increase the currentFocus variable:*/
          currentFocus++;
          /*and and make the current item more visible:*/
          addActive(x);
        } else {
          showList(true);
        }
      } else if (e.keyCode == 38) {
        //up
        /*If the arrow UP key is pressed,
        decrease the currentFocus variable:*/
        currentFocus--;
        /*and and make the current item more visible:*/
        addActive(x);
      } else if (e.keyCode == 13) {
        /*If the ENTER key is pressed, prevent the form from being submitted,*/
        e.preventDefault();
        if (currentFocus > -1) {
          /*and simulate a click on the "active" item:*/
          if (x) x[currentFocus].click();
        }
      } else if (e.keyCode == 27) {
        closeAllLists();
      }
    });
    function showList(showAll) {
      if (showAll && isListOpen()) {
        closeAllLists();
        return;
      }

      let a,
        b,
        i,
        inputValue = inp.value;

      /*close any already open lists of autocompleted values*/
      closeAllLists();

      if (!showAll && !inputValue) {
        return false;
      }
      currentFocus = -1;
      /*create a DIV element that will contain the items (values):*/
      a = document.createElement('DIV');
      a.setAttribute('id', inp.id + 'autocomplete-list');
      a.setAttribute('class', 'autocomplete-items');
      /*append the DIV element as a child of the autocomplete container:*/
      inp.parentNode.appendChild(a);
      /*for each item in the array...*/
      for (i = 0; i < src.length; i++) {
        /*check if the item starts with the same letters as the text field value:*/
        const srcObj = src[i];
        const srcValue = srcObj[valueField];
        const srcDisplay = srcObj[displayField];
        if (
          srcDisplay.toUpperCase().includes(inputValue.toUpperCase()) ||
          showAll
        ) {
          /*create a DIV element for each matching element:*/
          b = document.createElement('DIV');
          if (showAll) {
            if (srcDisplay.toUpperCase() === inputValue.toUpperCase()) {
              b.innerHTML = '<strong>' + srcDisplay + '</strong>';
            } else {
              b.innerHTML = srcDisplay;
            }
          } else {
            /*make the matching letters bold:*/
            const valIndex = srcDisplay
              .toUpperCase()
              .indexOf(inputValue.toUpperCase());
            b.innerHTML = srcDisplay.substr(0, valIndex);
            b.innerHTML +=
              '<strong>' +
              srcDisplay.substr(valIndex, inputValue.length) +
              '</strong>';
            b.innerHTML += srcDisplay.substr(valIndex + inputValue.length);
          }

          /*insert a input field that will hold the current array item's value:*/
          b.innerHTML += "<input type='hidden' value='" + srcValue + "'>";
          /*execute a function when someone clicks on the item value (DIV element):*/
          b.addEventListener('click', function(e) {
            /*insert the value for the autocomplete text field:*/
            const value = this.getElementsByTagName('input')[0].value;
            const selectedItem = self.props.source.find(
              itm => itm[valueField] + '' === value + ''
            );
            const display = selectedItem
              ? selectedItem[self.props.displayField]
              : '';

            // inp.value = display;
            // self.state.value[displayField] = display;

            // valInp.value = value;
            /*close the list of autocompleted values,
            (or any other open lists of autocompleted values:*/
            closeAllLists();
            onChange(value);
          });
          a.appendChild(b);
        }
      }
    }
    function addActive(x) {
      /*a function to classify an item as "active":*/
      if (!x) return false;
      /*start by removing the "active" class on all items:*/
      removeActive(x);
      if (currentFocus >= x.length) currentFocus = 0;
      if (currentFocus < 0) currentFocus = x.length - 1;
      /*add class "autocomplete-active":*/
      x[currentFocus].classList.add('autocomplete-active');
    }
    function removeActive(x) {
      /*a function to remove the "active" class from all autocomplete items:*/
      for (var i = 0; i < x.length; i++) {
        x[i].classList.remove('autocomplete-active');
      }
    }
    function closeAllLists(elmnt) {
      /*close all autocomplete lists in the document,
      except the one passed as an argument:*/
      let x = inp.parentNode.getElementsByClassName('autocomplete-items');
      for (var i = 0; i < x.length; i++) {
        if (elmnt != x[i] && elmnt != inp) {
          x[i].parentNode.removeChild(x[i]);
        }
      }
    }
    function isListOpen() {
      return !!inp.parentNode.querySelector('.autocomplete-items');
    }
    /*execute a function when someone clicks in the document:*/
    document.addEventListener('click', function(e) {
      closeAllLists(e.target);
    });
  }

  // <div className="backline" />

  render() {
    return (
      <div className="autocomplete">
        <input
          id={this.props.id + '-text'}
          type="text"
          className={this.props.className}
          placeholder={this.props.placeholder}
          autoComplete="off"
        />
        <input id={this.props.id + '-hidden'} type="hidden" />
      </div>
    );
  }
}
