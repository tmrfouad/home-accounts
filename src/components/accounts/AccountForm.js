import React from 'react';
import { connect } from 'react-redux';
import {
  startAddAccount,
  startEditAccount,
  startRemoveAccount
} from '../../actions/accounts';
import ConfirmModal from '../ConfirmModal';
import { BackButton } from '../BackButton';

export class AccountForm extends React.Component {
  constructor(props) {
    super(props);
    const account = this.props.account;
    this.state = {
      name: account ? account.name : '',
      error: '',
      mode: account ? 'edit' : 'add',
      isModalOpen: false
    };
  }

  onNameChange = e => {
    this.setState({ name: e.target.value });
  };

  onFormSubmit = e => {
    e.preventDefault();
    if (!this.state.name) {
      this.setState({ error: 'Please enter a name!' });
    } else {
      if (this.state.mode === 'edit') {
        this.props.startEditAccount(this.props.account.id, {
          name: this.state.name
        });
      } else {
        if (this.props.accounts.find(acc => acc.name === this.state.name)) {
          this.setState({ error: 'This account already exists!' });
        } else {
          this.props.startAddAccount({
            name: this.state.name
          });
        }
      }
      this.props.history.push('/accounts');
    }
  };

  removeAccount = () => {
    this.props.startRemoveAccount(this.props.account.id);
    this.props.history.push('/accounts');
  };

  openRemoveItemDialog = e => {
    e.preventDefault();
    this.setState(() => ({ isModalOpen: true }));
  };

  closeRemoveItemDialog = () => {
    this.setState(() => ({ isModalOpen: false }));
  };

  render() {
    return (
      <div>
        <div className="page-header">
          <div
            className={
              this.props.styles.collapsed
                ? 'content-container content-container--collapsed'
                : 'content-container'
            }
          >
            <BackButton />
            <h2 className="page-header__title">
              {this.state.mode === 'edit' ? 'Edit' : 'Add'} Account
            </h2>
          </div>
        </div>
        <form
          className={
            this.props.styles.collapsed
              ? 'form content-container content-container--collapsed'
              : 'form content-container'
          }
          onSubmit={this.onFormSubmit}
        >
          {this.state.error && (
            <div className="form__error">{this.state.error}</div>
          )}
          <input
            type="text"
            className="text-input"
            autoFocus
            value={this.state.name}
            onChange={this.onNameChange}
          />
          <div>
            <button className="button form__action">Save Account</button>
            {this.state.mode === 'edit' && (
              <button
                className="button form__action button--secondary"
                onClick={this.openRemoveItemDialog}
              >
                Remove Account
              </button>
            )}
          </div>
        </form>
        <ConfirmModal
          id="confirmModal"
          messageTitle="Remove Account!"
          messageBody="Are you sure you want to remove this item?"
          isOpen={this.state.isModalOpen}
          onModalClose={this.closeRemoveItemDialog}
          onModalOk={this.removeAccount}
        />
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  account: state.accounts.find(t => t.id === props.match.params.id),
  accounts: state.accounts,
  styles: state.styles
});

const mapDispatchToProps = dispatch => ({
  startAddAccount: account => dispatch(startAddAccount(account)),
  startEditAccount: (id, account) =>
    dispatch(startEditAccount({ id }, account)),
  startRemoveAccount: id => dispatch(startRemoveAccount({ id }))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AccountForm);
