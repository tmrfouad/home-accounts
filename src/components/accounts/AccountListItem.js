import React from 'react';
import { Link } from 'react-router-dom';

const AccountListItem = ({ id, name }) => (
  <Link className="list-item" to={`/accountform/${id}`}>
    <div>
      <h3 className="list-item__title">{name}</h3>
    </div>
  </Link>
);

export default AccountListItem;
