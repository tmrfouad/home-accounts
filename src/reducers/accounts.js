// Accounts Reducer

const accountsReducerDefaultState = [];

export default (state = accountsReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_ACCOUNT':
      return [...state, action.account];
    case 'REMOVE_ACCOUNT':
      return state.filter(({ id }) => id !== action.id);
    case 'EDIT_ACCOUNT':
      return state.map(
        account =>
          account.id === action.id ? { ...account, ...action.updates } : account
      );
    case 'SET_ACCOUNTS':
      return action.accounts;
    default:
      return state;
  }
};
