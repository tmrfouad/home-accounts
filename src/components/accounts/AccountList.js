import React from 'react';
import { connect } from 'react-redux';
import AccountListItem from './AccountListItem';
import { Link } from 'react-router-dom';

export class AccountList extends React.Component {
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h2 className="page-header__title">Accounts</h2>
            <div className="page-header__actions">
              <Link className="button" to="/accountform">
                Add Account
              </Link>
            </div>
          </div>
        </div>
        <div className="content-container">
          <div className="list-header">
            <div>Name</div>
          </div>
          <div className="list-body">
            {this.props.accounts.length === 0 ? (
              <div className="list-item list-item--message">
                <span>No Accounts</span>
              </div>
            ) : (
              this.props.accounts.map(acc => (
                <AccountListItem key={acc.id} {...acc} />
              ))
            )}
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  accounts: state.accounts
});

export default connect(mapStateToProps)(AccountList);
