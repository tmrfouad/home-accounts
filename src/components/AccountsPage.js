import React from 'react';
import AccountList from './AccountList';

const AccountsPage = () => (
  <div>
    <AccountList />
  </div>
);

export default AccountsPage;
