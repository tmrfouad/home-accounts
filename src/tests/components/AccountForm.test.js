import React from 'react';
import { shallow } from 'enzyme';
import { AccountForm } from '../../components/AccountForm';
import accounts from '../fixtures/accounts';

let wrapper, startAddAccount, startEditAccount, startRemoveAccount, history;

beforeEach(() => {
  startAddAccount = jest.fn();
  startEditAccount = jest.fn();
  startRemoveAccount = jest.fn();
  history = { push: jest.fn() };
  wrapper = shallow(
    <AccountForm
      startAddAccount={startAddAccount}
      startEditAccount={startEditAccount}
      startRemoveAccount={startRemoveAccount}
      history={history}
      accounts={accounts}
    />
  );
});

test('should render account form correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render account form with expense data', () => {
  wrapper.setProps({
    account: accounts[0]
  });
  expect(wrapper).toMatchSnapshot();
});

test('should render error for invalid form submission', () => {
  expect(wrapper).toMatchSnapshot();
  wrapper.find('form').simulate('submit', { preventDefault: () => {} });
  expect(wrapper.state('error').length).toBeGreaterThan(0);
  expect(wrapper).toMatchSnapshot();
});

test('should set name on input change', () => {
  const value = 'Account 1 name';
  wrapper
    .find('input')
    .at(0)
    .simulate('change', { target: { value: value } });
  expect(wrapper.state('name')).toBe(value);
});

test('should handle startAddExpense', () => {
  expect(wrapper.state('mode')).toBe('add');

  const name = 'New Account';
  wrapper.setState({ name });

  wrapper.find('form').prop('onSubmit')({
    preventDefault: () => {}
  });

  expect(wrapper.state('error')).toBe('');
  expect(startAddAccount).toHaveBeenLastCalledWith({
    name: name
  });
  expect(history.push).toHaveBeenLastCalledWith('/accounts');
});

test('should handle startEditExpense', () => {
  wrapper = shallow(
    <AccountForm
      startEditAccount={startEditAccount}
      history={history}
      accounts={accounts}
      account={accounts[0]}
    />
  );

  expect(wrapper.state('mode')).toBe('edit');

  wrapper.find('form').prop('onSubmit')({
    preventDefault: () => {}
  });

  expect(wrapper.state('error')).toBe('');
  expect(startEditAccount).toHaveBeenLastCalledWith(accounts[0].id, {
    name: accounts[0].name
  });
  expect(history.push).toHaveBeenLastCalledWith('/accounts');
});

test('should handle startRemoveExpense', () => {
  wrapper = shallow(
    <AccountForm
      startRemoveAccount={startRemoveAccount}
      history={history}
      accounts={accounts}
      account={accounts[0]}
    />
  );

  expect(wrapper.state('mode')).toBe('edit');

  wrapper
    .find('button')
    .at(1)
    .prop('onClick')({
    preventDefault: () => {}
  });

  wrapper.find('#confirmModal').prop('onModalOk')();

  expect(wrapper.state('error')).toBe('');
  expect(startRemoveAccount).toHaveBeenLastCalledWith(accounts[0].id);
  expect(history.push).toHaveBeenLastCalledWith('/accounts');
});
