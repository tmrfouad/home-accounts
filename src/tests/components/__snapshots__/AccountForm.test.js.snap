// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render account form correctly 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h2
        className="page-header__title"
      >
        Add
         Account
      </h2>
    </div>
  </div>
  <form
    className="form content-container"
    onSubmit={[Function]}
  >
    <input
      className="text-input"
      onChange={[Function]}
      type="text"
      value=""
    />
    <div>
      <button
        className="button form__action"
      >
        Save Account
      </button>
      <button
        className="button form__action button--secondary"
        onClick={[Function]}
      >
        Remove Account
      </button>
    </div>
  </form>
  <ConfirmModal
    id="confirmModal"
    isOpen={false}
    messageBody="Are you sure you want to remove this item?"
    messageTitle="Remove Account!"
    onModalClose={[Function]}
    onModalOk={[Function]}
  />
</div>
`;

exports[`should render account form with expense data 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h2
        className="page-header__title"
      >
        Add
         Account
      </h2>
    </div>
  </div>
  <form
    className="form content-container"
    onSubmit={[Function]}
  >
    <input
      className="text-input"
      onChange={[Function]}
      type="text"
      value=""
    />
    <div>
      <button
        className="button form__action"
      >
        Save Account
      </button>
      <button
        className="button form__action button--secondary"
        onClick={[Function]}
      >
        Remove Account
      </button>
    </div>
  </form>
  <ConfirmModal
    id="confirmModal"
    isOpen={false}
    messageBody="Are you sure you want to remove this item?"
    messageTitle="Remove Account!"
    onModalClose={[Function]}
    onModalOk={[Function]}
  />
</div>
`;

exports[`should render error for invalid form submission 1`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h2
        className="page-header__title"
      >
        Add
         Account
      </h2>
    </div>
  </div>
  <form
    className="form content-container"
    onSubmit={[Function]}
  >
    <input
      className="text-input"
      onChange={[Function]}
      type="text"
      value=""
    />
    <div>
      <button
        className="button form__action"
      >
        Save Account
      </button>
      <button
        className="button form__action button--secondary"
        onClick={[Function]}
      >
        Remove Account
      </button>
    </div>
  </form>
  <ConfirmModal
    id="confirmModal"
    isOpen={false}
    messageBody="Are you sure you want to remove this item?"
    messageTitle="Remove Account!"
    onModalClose={[Function]}
    onModalOk={[Function]}
  />
</div>
`;

exports[`should render error for invalid form submission 2`] = `
<div>
  <div
    className="page-header"
  >
    <div
      className="content-container"
    >
      <h2
        className="page-header__title"
      >
        Add
         Account
      </h2>
    </div>
  </div>
  <form
    className="form content-container"
    onSubmit={[Function]}
  >
    <div
      className="form__error"
    >
      Please enter a name!
    </div>
    <input
      className="text-input"
      onChange={[Function]}
      type="text"
      value=""
    />
    <div>
      <button
        className="button form__action"
      >
        Save Account
      </button>
      <button
        className="button form__action button--secondary"
        onClick={[Function]}
      >
        Remove Account
      </button>
    </div>
  </form>
  <ConfirmModal
    id="confirmModal"
    isOpen={false}
    messageBody="Are you sure you want to remove this item?"
    messageTitle="Remove Account!"
    onModalClose={[Function]}
    onModalOk={[Function]}
  />
</div>
`;
